theano/tensor/nlinalg.py

```diff
@@ -726,3 +726,59 @@ def norm(x, ord):
             raise ValueError(0)
     elif ndim > 2:
         raise NotImplementedError("We don't support norm witn ndim > 2")
+
+
+class TensorInv(Op):
+    """
+    Class wrapper for tensorinv() function;
+    Theano utilization of numpy.linalg.tensorinv;
+    """
+    _numop = staticmethod(numpy.linalg.tensorinv)
+    __props__ = ('ind',)
+
+    def __init__(self, ind=2):
+        self.ind = ind
+
+    def make_node(self, a):
+        a = as_tensor_variable(a)
+        out_dtype = a.dtype
+        out = theano.tensor.matrix(dtype=out_dtype)
+        return Apply(self, [a], [out])
+
+    def perform(self, node, inputs, outputs):
```

Comment:
Can you add the infer_shape method to that class and associated tests? Not having it can causes unexpected behavior to the user.